x-pack/legacy/plugins/integrations_manager/server/packages/fields.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { readFileSync } from 'fs';
+import { safeLoad } from 'js-yaml';
+
+// This should become a copy of https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/mapping/field.go#L39
+export interface Field {
+  keyword: string;
+  type: string;
+  required: boolean;
+  description: string;
+  fields: Field[];
+}
+
+test('test reading fields.yml', () => {
+  const yaml = readFileSync(__dirname + '/tests/fields.yml');
+  const data = safeLoad(yaml.toString());
+
+  console.log(keyword());
+  data.forEach(data => {
+    console.log(data as Field);
+  });
+  expect(1 + 2).toStrictEqual(3);
+});
+
+function getDefaultProperties() {
+  // TODO: do be extended with https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/template/processor.go#L364
+  // Currently no defaults exist
+  return {};
+}
+
+function keyword() {
+  const property = getDefaultProperties();
```

Comment:
Another option is 

```suggestion
  const property = Object.assign(getDefaultProperties(), {type: 'keyword'});
```

Reply:
There will be quite a few more fields which will be added to the properties object but with if / else conditions. So I wonder if it is discouraged in `js` to use the dot notation to do this directly?

Reply:
One situation where this is necessary is if you type the object (in TypeScript) and the properties are mandatory. Then `Object.assign()` is a good way to prefill everything in one go. Other than that, I don't think so.